packages/ripple-binary-codec/src/binary.ts

```diff
@@ -16,26 +20,43 @@ import * as bigInt from 'big-integer'
  * Construct a BinaryParser
  *
  * @param bytes hex-string to construct BinaryParser from
+ * @param definitions rippled definitions used to parse the values of transaction types and such.
+ *                          Can be customized for sidechains and amendments.
  * @returns A BinaryParser
  */
-const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes)
+const makeParser = (
+  bytes: string,
+  definitions?: XrplDefinitions,
+): BinaryParser => new BinaryParser(bytes, definitions)
 
 /**
  * Parse BinaryParser into JSON
  *
  * @param parser BinaryParser object
+ * @param definitions rippled definitions used to parse the values of transaction types and such.
+ *                          Can be customized for sidechains and amendments.
  * @returns JSON for the bytes in the BinaryParser
  */
-const readJSON = (parser: BinaryParser): JsonObject =>
-  (parser.readType(coreTypes.STObject) as STObject).toJSON()
+const readJSON = (
+  parser: BinaryParser,
+  definitions: XrplDefinitions = DEFAULT_DEFINITIONS,
```

Comment:
Why is there a default `definitions` provided in some methods but not all?

Reply:
I removed explicit parameters from all but the bottom functions. So if they only use functions which have `DEFAULT_DEFINITIONS` defined, then I just had them use the default for that function. (This was in response to another comment earlier where it seemed better to avoid over-specifying which definitions to use at the top level)

I'm honestly not sure what's better. My thought process is that it probably makes sense to avoid explicit defaults across package boundaries (So having explicit default of `DEFAULT_DEFINITIONS` within the binary codec, but not having those within xrpl.js functions) in order to allow the lower-level package to update that default independently of updating xrpl.js. So maybe I should add the explicit default for all the functions within the binary codec which use this param?